algolia/search/client.go

```diff
@@ -127,3 +127,26 @@ func (c *Client) CustomRequest(
 ) error {
 	return c.transport.Request(&res, method, path, body, k, opts...)
 }
+
+// GetStatus retrieves the task status according to the Algolia engine for the
+// given task.
+func (c *Client) GetStatus(taskID int64, opts ...interface{}) (res TaskStatusRes, err error) {
+	path := c.path("/task/%d", taskID)
+	err = c.transport.Request(&res, http.MethodGet, path, nil, call.Read, opts...)
+	return
+}
+
+// WaitTask blocks until the task identified by the given taskID is completed on
+// Algolia engine.
+func (c *Client) WaitTask(taskID int64, opts ...interface{}) error {
+	return waitWithRetry(func() (bool, error) {
+		res, err := c.GetStatus(taskID, opts...)
+		if err != nil {
+			return true, err
```

Comment:
Shouldn't this be better off as `false, err` ? Again, maybe it's similar to other impls, in which case please just disregard.

Reply:
Yes, this logic is a copy-paste of `waitWithRetry` for the index. I suppose this was done to not retry inside `waitWithRetry` but to return an error and proceed retrying on the upper level with a different host.